frontend/gist-frontend/src/api/api-handler.ts

```diff
@@ -3,27 +3,35 @@ import { local_url } from "./urls";
 export async function APIHandler(
   extension: string,
   method: string,
-  body?: object
+  body?: object,
+  info?: string
 ) {
   let response = "";
-  if (method === "GET") {
-    await fetch(`${local_url}/${extension}`)
+  let data = "";
+  if (!!body) {
+    data = JSON.stringify(body);
+  }
+  if (method === "POST") {
+    await fetch(`${local_url}/${extension}`, {
+      method: method,
+      body: data,
+      headers: {
+        "Content-Type": "application/json"
+      }
+    })
       .then(data => {
         return data.json();
       })
       .then(json => {
         response = json;
       })
       .catch(console.log);
+
+    console.log(response);
     return response;
-  } else if (method === "POST") {
-    let data = JSON.stringify(body);
+  } else {
```

Comment:
I'm kinda curious as to what's going on here towards the bottom. I'm noticing a conditional check for only POST and I'm curious as to why it's just the one check. I'm not sure if you're just not finished or have some master plan here, but I'm not sure how this works for other method types besides POST; however, I'm probably just not reading deep enough into this.

Reply:
POST requests have a body so it has to be included, whereas GET can't have a body. I tried putting them all in one but the body part was screwing it up. Probably going to be changed later to see if I can get them all in one, but for now this was how i did it.